frontend/components/common/CompatibleDevice.tsx

```diff
@@ -1,11 +1,22 @@
-import { Flex, Img, Text } from '@chakra-ui/react';
+import { Flex, Img, Link, Text } from '@chakra-ui/react';
+import NextLink from 'next/link';
 import { Product } from '@models/product';
 
 export type CompatibleDeviceProps = {
    device: NonNullable<Product['compatibility']>['devices'][number];
+   truncate?: number;
 };
 
-export function CompatibleDevice({ device }: CompatibleDeviceProps) {
+export function CompatibleDevice({
+   device,
+   truncate = 0,
+}: CompatibleDeviceProps) {
+   let variants = device.variants;
+   let unlistedVariants = 0;
+   if (truncate > 0 && variants.length > truncate) {
```

Comment:
Based on Sterling's comment https://github.com/iFixit/react-commerce/pull/935#issuecomment-1295396340, I'd expect this to look like `variants.length > truncate + 1` to handle the case where we only have one extra variant. Maybe I'm missing something and it's being handled elsewhere.

Reply:
im not quite sure what you mean, my solution is just maxed at `truncate` variants, whiles sterlings is `truncate` + 1 variants.  I felt like `truncate` being the cap made more intuitive sense.

Reply:
Oh I see, `truncate` doesn't indicate how many variants to show _after_ truncation, it indicates the max number of variants to show before we start truncating.